new-docs/content/any/getting-started/policies/index.md

```diff
@@ -9,325 +9,401 @@ description: Learn about writing Oso policies - the source of truth for authoriz
-the same logic on an unspecialized rule:
+```polar
+allow(actor: User, "GET", resource: Expense) if ...;
+```
 
-{{< code file="policy.polar" >}}
-allow(actor, action, resource) if actor matches User{name: "alice"};
-{{< /code >}}
+### Built-in Classes
 
-Either way, using the rule could look like this:
+We said above that any class you wish to use as a specializer
+(or on the right-hand side of the `matches` operator) must be
+registered with Polar. For convenience, Polar automatically
+registers a few classes, such as:
 
-{{% exampleGet "specializedExample" %}}
+* `Dictionary`
+* `String`
+* `List`
+* `Integer`
+* `Float`
 
-{{% callout "Note" "green" %}}
-Type specializers automatically respect the **inheritance** hierarchy of
-application classes. See the [Resources with
-Inheritance](guides/more/inheritance) guide for an in-depth
-example of how this works.
-{{% /callout %}}
+These classes correspond to ones in the application language,
+e.g., in Python, `String` is actually the built-in `str` class.
 
-Once a class is registered, class or static methods can also be called from Oso
-policies:
+## Method Calls
 
-{{< code file="policy.polar" >}}
-allow(actor: User, action, resource) if actor.name in User.superusers();
-{{< /code >}}
+In addition to accessing fields in a rule, you can also call methods
+on application objects. We'll demonstrate this using the built-in
+`String` class, but it applies to instances any registered class.
 
-{{% exampleGet "classMethodExample" %}}
+Suppose we decided to change our encoding of actions so that
+every "read" action started with the letter "r". We could allow
+all such actions using a rule like this:
 
-### Built-in Types
+```polar
+allow(_actor, action: String, _resource) if
+    action.startswith("r");
+```
 
-Methods called on the Polar built-in types `String`, `Dictionary`, `Number`,
-and `List` punt to methods on the corresponding application language class.
-That way you can use familiar methods like `.{{% exampleGet "startswith" %}}()`
-on strings regardless of whether they originated in your application or as a
-literal in your policy. This applies to all of Polar's [supported
-types](polar-syntax#primitive-types) in any supported application language. For
-examples using built-in types, see [the {{% exampleGet "langName" %}}
-library](reference/polar/classes) guide.
+## Other Rules
 
-{{% callout "Warning" "orange" %}}
-Do not attempt to mutate a literal using a method on it. Literals in Polar
-are constant, and any changes made to such objects by calling a method will
-not be persisted.
-{{% /callout %}}
+So far all of our examples have involved only `allow` rules
+with three parameters. That's a natural place to start with Oso,
+because the `is_allowed` function generates queries of the form
+`allow(actor, action, resource)`. But you can write rules for
+whatever you like, and query them from within your `allow` rules.
```

Comment:
This is a great explanation, and we should keep it here. But I think as mentioned in my earlier comment, having something similar to this when we first introduce `allow` rules, to foreshadow this section, would be great.